enhancements/catalogsource-config.md

```diff
@@ -0,0 +1,212 @@
+---
+title: catalogsource-config
+authors:
+  - "@perdasilva"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2021-11-18
+last-updated: 2021-11-18
+status: provisional
+---
+
+# catalogsource-config
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+
+## Summary
+
+Currently, there is no way to instruct OLM to schedule pods created from `CatalogSource` CRs on specific nodes.
+This is desirable for admins as it gives them more control over where in their infrastructure the catalog source
+pods will be created. In this enhancement proposal we
+propose changes to the `CatalogSource` spec to enable this kind customization possible for `grpc` type CatalogSources 
+that define a `spec.image` - the only catalog source configuration that produces a pod. 
+
+
+## Motivation
+
+This EP targets issue [#2452](https://github.com/operator-framework/operator-lifecycle-manager/issues/2452).
+Users would like to be able to override the pod spec of the CatalogSource pod
+created for `grpc` `CatalogSource`s. Specifically, they would like to be able to specify
+pod tolerations and node selectors.
```

Comment:
perhaps also worth mentioning https://github.com/operator-framework/operator-lifecycle-manager/issues/2666 as this enhancement would also give cluster autoscaler users a mechanism to enable using both components on the same cluster without issue.